Glossary article

randyscansani opened the pull request:
# Description
Hi! I drafted the article on glossaries under the parent folder "Customisation". Let me know you're feedback!

## Type of PR

- Creates the article Glossaries



### Checklist:

- [ ] I have read the [contributing guidelines](contributing.md).
- [ ] I have followed the [style guide](http://machinetranslate.org/style).
- [ ] I have made sure that the URL is not already in use.
- [ ] I have cross-linked relevant articles.
- [ ] I have used the UK spelling.
- [ ] I have kept consistency with the structure of sister articles in the same directory.

cefoo commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,27 @@ layout: coming_soon
+Adding a glossary to a commercial MT provider can increase the terminology accuracy of the output.
+However, the insertion of a term in the output happens regardless of the context.
+The terms included in the glossary should therefore be unambiguous, meaning that their translation should not be heavily dependent on the context.
+For example, "respiratory syndrome" is a good candidate to appear in a glossary, whereas the translation of the word "spread" may differ when it is used as a verb or a noun, or when the term occurs in the medical or in the financial domain.
```
What do you think of creating a section called `Examples` as we have [here](https://machinetranslate.org/language-model)?

cefoo commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,27 @@ layout: coming_soon
+
+The target term can be inserted either with the same form as in the glossary, or by adapting the term to the rest of the sentence and viceversa, for example ensuring a correct agreement between the subject and the verb.
+
+## Term forms to be included in a glossary for MT
```
Maybe... ? 

> ## Format

cefoo commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,27 @@ layout: coming_soon
+Some providers match the source term regardless of the morphology and ortography of its occurrence in the source sentence.
+
+
+The target term can be inserted either with the same form as in the glossary, or by adapting the term to the rest of the sentence and viceversa, for example ensuring a correct agreement between the subject and the verb.
```
What do you think of adding a `Format` section before this paragraph? We can omit the following title and make this paragraph and the next part of the same section.

randyscansani replied:
Agreed, the article needed some reorganization. I'm not 100% sure of the name "Format". When I think about the format of a glossary I think of tsv, tbx, csv and so on :) I've chosen "Glossary term forms". Let me know if you're ok with this!

cefoo replied:
Yeah, you are absolutely right.

bittlingmayer replied:
I suggest the same as for matching.

> Strict exact insertion is a problem because ...
>
> In some machine translation systems, term insertion can handle common variants, according to the context.
> - Morphological agreement
> - Uppercase and lowercase
> ...

cefoo commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,27 @@ layout: coming_soon
+
+## Term forms to be included in a glossary for MT
+If the source and target terms are used with the same form as in the glossary, the latter should cover all possible morphological and ortographical forms of each term.
+If the term is matched regardless of its form and is inflected, if needed, so that it agrees with the rest of the items in the sentence, it is possible to include in the glossary only the dictionary form of each term.
```
I'm thinking that we may include a list of the possible term varieties. This is my suggestion, feel free to adapt it as you would like:

If the machine translation system inserts glossary terms literally, the glossary should include all possible term varieties, depending on the language:

- Morphological forms
    - Singular and plural forms
    - Female and male inflections
    - Verb inflections 

If the machine translation system adapts glossary terms automatically, the glossary can only include the dictionary form of each term.

Lists are easier to scan than paragraphs.

cefoo reviewed (commented):
Hi Randy!
Thanks a lot for the PR!
I'll insert the following lists:
- Commercial MT systems that support glossaries (and, if I find the info, whether they require all term forms or not)
- TMS and CAT tools where the user can select a glossary to customize the MT system

If you have any comment or suggestion, please feel free to let me know!
I know this article will be super useful for potential users. 😊

randyscansani commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,27 @@ layout: coming_soon
+## How glossaries are leveraged by MT systems
+Glossaries are leveraged at translation time, and no training is required.
+Adding a glossary to a commercial MT provider can increase the terminology accuracy of the output.
+However, the insertion of a term in the output happens regardless of the context.
```
I think here we are still talking about the "general" application of the glossary feature, which is common to any provider that implements it, so I would still keep this part under the "Application" section. I would start a new section when we talk about the differences between the implementations of the glossary features (source matching and target insertion). What do you think?

cefoo replied:
I think your final version reads very well!

randyscansani commented:
Hi Cecilia,

many thanks for your feedback!! I've tried and re-organized the article a little bit based on your suggestions. Hope it looks better now!
Regarding the items you would like to add, please find a couple of notes based on my experience :)

- _Commercial MT systems that support glossaries (and, if I find the info, whether they require all term forms or not)_ It's really hard to find such details because in general providers do not disclose how the glossary feature works, so often you need to test and you cannot use the official documentation as a reference in the article. Based on previous tests, I can tell you that Google (https://cloud.google.com/translate/docs/advanced/glossary) works with exact matches, whereas DeepL matches terms on a lemma level and inflects the target. Microsoft provides different types of glossary features, some of which are more advances, while others work as Google (https://docs.microsoft.com/en-us/azure/cognitive-services/translator/custom-translator/what-is-dictionary). I don't know much about the Amazon feature, unfortunately. 

- _TMS and CAT tools where the user can select a glossary to customize the MT system_ I know about Memsource and Intento having their own glossary feature. Here it will be important to make a distinction between TMS and CAT tools that integrate third-party glossary features (e.g. via the DeepL plugin) and TMs and CAT tools that have developed their own feature.

cefoo commented:
Thanks for providing such valuable info, Randy! ✨

I am currently working on multiple data files that will allow automatically generating sections in articles `Languages`, `Features`, `TMS integrations` and, as we've been discussing, `Glossary`.
You can check them out [here](https://github.com/machinetranslate/machinetranslate.org/pull/263) and [here](https://github.com/machinetranslate/machinetranslate.org/pull/269).

randyscansani commented:
That's really cool! Thanks Cecilia

bittlingmayer commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+## Application
+
+Glossaries are leveraged at translation time, and no training is required.
+When a glossary term is matched in the source sentence, its translation is enforced in the output.
```
@cefoo Maybe we should decide site-wide about this?

"Sequence" seems the most neutral (the way the human translation industry has settled on "segment").

bittlingmayer replied:
And then we define it somewhere.

cefoo replied:
Vilém created the `string` article, where he defines `string` as `a sequence of characters`.
Perhaps we can use this term and hyperlink it to that article.

bittlingmayer commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+##Term matching and insertion
+
+Several machine translation providers offer the possibility to enhance their output using a glossary.
+The way in which the glossary is leveraged may differ in terms of source term matching and target term insertion.
```
> Machine translation systems differ in how they match and insert glossary terms.

bittlingmayer commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+Several machine translation providers offer the possibility to enhance their output using a glossary.
+The way in which the glossary is leveraged may differ in terms of source term matching and target term insertion.
+
+In some cases the source term is matched in the source text only if it occurs with the same morphology and ortography as in the glossary.
```
What do you think about something like this?

> Approaches to term matching:
> - Strict exact term matching: ...
> - Morphology-aware term matching: ...
> - 

Or 

> Strict exact term matching is a problem because terms can have varying forms.
> 
> With some machine translation systems, term matching can handle common variants.
> - Morphology
> - Orthography

By orthography here do you mean like American/British spelling differences, or more?  We could replace orthography with the actual issues.

> - Spelling variants
> - Uppercase and lowercase

bittlingmayer commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+
+## Glossary term forms
+
+If the source and target terms are used with the same form as in the glossary, the latter should cover the following forms, depending on the language:
```
I think this section a bit too much detail, and it's very related to the previous section.

We could just replace it all with:

> With strict exact matching and insertion, more variants need to be listed in the glossary.

Or

> With matching and insertion that handles common variants, multiple variants of the same term do not need to be listed in the glossary.

What do you think?

randyscansani replied:
Agreed!

randyscansani commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+
+##Term matching and insertion
+
+Several machine translation providers offer the possibility to enhance their output using a glossary.
```
Good question. We can't be 100% sure that this is true because many providers do not disclose how the glosary functionality works, so if we try to be more specific, there is a risk that what we say does not apply to (all of?) them. WHat do you think?

bittlingmayer replied:
Then we should remove this whole section or, if we feel it's common enough (I do), just make it

> Glossaries can be applied in two steps:

So "are" -> "can be"

randyscansani commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
 title: Glossaries
 description: Terminology lists for training and customising machine translation
```
Actually Microsoft offers a functionality called "Dictionary-only training" where you launch a "training" with a glossary only. https://docs.microsoft.com/en-us/azure/cognitive-services/translator/custom-translator/what-is-dictionary#dictionary-only-trainings
I guess it's some sort of fine-tuning of their generic model with a glossary, but since they call it "training" I was wondering whether it makes sense that "training" appears in the title? Perhaps that's why it was there in the first place?

bittlingmayer replied:
Interesting, good to know

"customizing" is more general so it is true for all the engines.

It lets us avoid asserting whether model training happens or not.

randyscansani commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
+#Glossaries
+
+Glossaries are lists of terms in one language aligned to their translation in one or more languages. They can be added to machine translation systems.
+Adding a glossary to a commercial machine translation provider can increase the terminology accuracy of the output.
```
I tried to rephrase it and insert the bullet list

randyscansani commented:
Hi! I've tried to implement Adam's suggestions. Let me know :)

cefoo commented on customisation/glossaries.md:
```diff
@@ -5,3 +5,45 @@ layout: coming_soon
 title: Glossaries
 description: Terminology lists for training and customising machine translation
 ---
+#Glossaries
```
Done

bittlingmayer commented:
I made a PR with some suggestions, easier than commenting:

ttps://github.com/randyscansani/machinetranslate.org/pull/1

randyscansani commented:
I merged Adam's edits and aplied a couple of things we had already said: removed the title and replaced "sentence" with "string" with the ref to the article. Thanks!
